ui/src/app/pages/workspace/workspace-edit-validation.ts

```diff
@@ -31,63 +41,62 @@ export type WorkspaceEditValidationArgs = Required<
+        z.string(),
+        "can't be blank"
+      ),
+      primaryPurpose: trueBoolean(),
 
-  // TODO: This validation spec should include error messages which get
-  // surfaced directly. Currently these constraints are entirely separate
-  // from the user facing error strings we render.
-  const constraints: object = {
-    name: requiredStringWithMaxLength(80, 'Name'),
-    // The prefix for these lengthMessages require HTML formatting
-    // The prefix string is omitted here and included in the React template below
-    billingAccountName: { presence: true },
-    intendedStudy: requiredStringWithMaxLength(1000),
-    populationChecked: { presence: true },
-    anticipatedFindings: requiredStringWithMaxLength(1000),
-    reviewRequested: { presence: true },
-    scientificApproach: requiredStringWithMaxLength(1000),
-    researchOutcomeList: { presence: { allowEmpty: false } },
-    disseminateResearchFindingList: { presence: { allowEmpty: false } },
-    primaryPurpose: { truthiness: true },
-
-    // Conditionally include optional fields for validation.
-    otherPurposeDetails: values.otherPurpose
-      ? requiredStringWithMaxLength(500, 'Other primary purpose')
-      : {},
-    populationDetails: values.populationChecked ? { presence: true } : {},
-    otherPopulationDetails: values.populationDetails?.includes(
-      SpecificPopulationEnum.OTHER
-    )
-      ? requiredStringWithMaxLength(100, 'Other Specific Population')
-      : {},
-    diseaseOfFocus: values.diseaseFocusedResearch
-      ? requiredStringWithMaxLength(80, 'Disease of Focus')
-      : {},
-    otherDisseminateResearchFindings:
-      values.disseminateResearchFindingList?.includes(
-        DisseminateResearchEnum.OTHER
+      // Optional fields that are required based on conditions
+      // old validation allowed for "present" but empty array [] value.
+      otherPurposeDetails: data.otherPurpose
+        ? requiredStringWithMaxLength(500, 'Other primary purpose')
+        : noop,
+      populationDetails: data.populationChecked
+        ? nonEmptyArray(z.string(), "can't be blank")
+        : noop,
+      otherPopulationDetails: data.populationDetails?.includes(
+        SpecificPopulationEnum.OTHER
       )
-        ? requiredStringWithMaxLength(
-            100,
-            'Other methods of disseminating research findings'
-          )
-        : {},
-    aianResearchType: askAboutAIAN ? { presence: true } : {},
-    aianResearchDetails: askAboutAIAN ? requiredStringWithMaxLength(1000) : {},
-  };
+        ? requiredStringWithMaxLength(100, 'Other Specific Population')
+        : noop,
+      diseaseOfFocus: data.diseaseFocusedResearch
+        ? requiredStringWithMaxLength(80, 'Disease of Focus')
+        : noop,
+      otherDisseminateResearchFindings:
+        data.disseminateResearchFindingList?.includes(
+          DisseminateResearchEnum.OTHER
+        )
+          ? requiredStringWithMaxLength(
```

Comment:
![image](https://github.com/user-attachments/assets/ae45627c-6aca-49e3-b889-0f7321435635)

Reply:
![image](https://github.com/user-attachments/assets/be556bbf-5523-4f44-a730-21e052ddacf3)

Reply:
When I have a large value in this field and it is checked, I get the error. If I uncheck the other dissemination box, the contents stay the same, I don't get the error, and I can submit.

I am in the process of seeing if submitting a workspace in this situation will break anything.

Reply:
The workspace creation errored out.